web/packages/teleport/src/Roles/RoleEditor/StandardEditor/AdminRules.tsx

```diff
@@ -194,3 +212,117 @@ function ResourceKindMultiValue(props: MultiValueProps<ResourceKindOption>) {
     </HoverTooltip>
   );
 }
+
+/** Renders a grid of allowed permissions (verbs) as a grid of checkboxes. */
+function VerbEditor({
+  verbs,
+  allVerbs,
+  rule,
+  onVerbChange,
+  onAllVerbsChange,
+}: {
+  verbs: VerbModel[];
+  allVerbs: boolean;
+  rule: Rule<VerbModel[]>;
+  onVerbChange(verb: Verb, checked: boolean): void;
+  onAllVerbsChange(checked: boolean): void;
+}) {
+  const id = useId();
+  const { valid, message } = useRule(rule(verbs));
+
+  // This is dumb, but it works for now, as the editor is fixed-width (defined
+  // in the RoleEditorAdapter component).
```

Comment:
suggestion: "Hardcoded column works here because the editor is fixed-width (defined in the RoleEditorAdapter component)"